### PR TITLE
Introduce PHPStan and fix level 0 problems

### PR DESCRIPTION
**Pull request by szepeviktor**
@acobster What do you think?

I know-I know. I've broken PHP 7.0 compat :(

### CODE REVIEW

**Comment by szepeviktor** [comment]
Strange that the Timber 2 branch contains `"timber/timber": "^1.7"` 🤔

File: lib/Conifer/Post/BlogPost.php [hunk 1]
```diff
@@ -53,7 +55,7 @@ public static function get_all_published_years() : array {
 _SQL_;
 
     // phpcs:ignore WordPress.WP.PreparedSQL.NotPrepared
-    return $wpdb->get_col( $prepared, ARRAY_A );
+    return $wpdb->get_col( $sql, ARRAY_A );
```
Comment (acobster) [inline lib/Conifer/Post/BlogPost.php:58]: lol

**Comment by acobster** [comment]
Thanks for working on this!

> Strange that the Timber 2 branch contains "timber/timber": "^1.7" 

Strange indeed. Once we get a Timber 2.x beta release we can switch over to that. In the meantime feel free to change this to `2.x-factories` which has the latest changes we need. Some tests will probably still fail for now. I've been testing this in a pretty ad-hoc way in a downstream project for now using the bleeding edge versions of Timber and Conifer which is why Conifer itself is going to need some love before too long. See also #131

I'm fine with requiring PHP 7.1 or even higher in the Conifer beta, since Timber 2.x is a breaking change anyway. We're probably going to use at least 7.1 on Timber.

**Comment by szepeviktor** [comment]
> I'm fine with requiring PHP 7.1 or even higher in the Conifer beta

Done :)

**Comment by szepeviktor** [comment]
I let you raise PHPStan level as commented in `phpstan.neon.dist`.

**Pull request merged**